### PR TITLE
Handling `ruleOfProcessingTokenPriceWS` and `percentDifferenceToShowByPriceOnWS` extensions in price service

### PR DESCRIPTION
**Pull request by TrGiLong**
## Description of the changes
- Moving rules to price service. We should handle it globally

### CODE REVIEW

**Review by zafar4aev — APPROVED** [review]

File: Packages/KeyAppKit/Sources/KeyAppBusiness/Price/PriceRule.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    func adjustValue(token: SomeToken, price: TokenPrice, fiat _: String) -> PriceRuleHandler {
+        if token.keyAppExtension.ruleOfProcessingTokenPriceWS == .byCountOfTokensValue {
+            return .next(TokenPrice(currencyCode: price.currencyCode, value: 1.0, token: token))
+        }
+
+        return .next(price)
+    }
+}
+
+// De-noise price by measure percentage difference. If the difference is in allowed range, the value
```
Comment (bigearsenal) [inline Packages/KeyAppKit/Sources/KeyAppBusiness/Price/PriceRule.swift]: 1. Complete the comment.
2. Rename to DepeggingPriceRule

File: Packages/KeyAppKit/Sources/KeyAppBusiness/Price/PriceService+Helper.swift [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import Foundation
```
Comment (bigearsenal) [inline Packages/KeyAppKit/Sources/KeyAppBusiness/Price/PriceService+Helper.swift]: PriceService+Extensions.swift

File: Packages/KeyAppKit/Sources/KeyAppBusiness/Price/PriceRule.swift [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import BigDecimal
+import Foundation
+import KeyAppKitCore
+
+internal enum PriceRuleHandler {
+    case next(TokenPrice?)
+    case stop(TokenPrice?)
```
Comment (bigearsenal) [inline Packages/KeyAppKit/Sources/KeyAppBusiness/Price/PriceRule.swift]: next -> `continue`
stop -> `break`

File: Packages/KeyAppKit/Tests/UnitTests/KeyAppBusinessTests/Price/PriceRuleTests.swift [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+            XCTAssertTrue(
+                newPrice?.token == token,
+                "The new price token should be the same instance as the input token"
+            )
+        case .stop:
+            XCTFail("Unexpected PriceRuleHandler.stop result")
+        }
+    }
+
+    func testDeNoisePriceRuleCase1() {
```
Comment (bigearsenal) [inline Packages/KeyAppKit/Tests/UnitTests/KeyAppBusinessTests/Price/PriceRuleTests.swift]: Denoise -> Depegging

**Review by bigearsenal — APPROVED** [review]

**Pull request merged**